phase08/StudentGrades/StudentGrades/DBController.cs

```diff
@@ -0,0 +1,46 @@
+using project;
+
+namespace DefaultNamespace;
+
+public class DBController
+{
+    private static DBController instance;
+
+    private DBController()
+    {
+    }
+
+    public static DBController Instance
+    {
+        get
+        {
+            if (instance == null) instance = new DBController();
+            return instance;
+        }
+    }
+
+    public void calculateAverageGrade()
+    {
+        using (var schoolContext = new SchoolDBContext())
+        {
+            foreach (var student in schoolContext.Students.ToList())
+            {
+                var studentAverage = schoolContext.StudentGrades.Where(x => x.StudentNumber == student.StudentNumber)
+                    .Average(x => x.Score);
+                student.AverageGrade = studentAverage;
+                schoolContext.Update(student);
+                schoolContext.SaveChanges();
+            }
+        }
+    }
+
+    public void AddRecordsToDB(List<Student> students, List<StudentGrade> lessons)
+    {
+        using (var schoolContext = new SchoolDBContext())
+        {
+            schoolContext.Students.AddRange(students);
+            schoolContext.StudentGrades.AddRange(lessons);
+            schoolContext.SaveChanges();
+        }
+    }
+}
```

Comment:
Not the best method because of 2 reasons:
1. You might only want to add new students or new lessons so for each call a new empty `List` must be instantiated.
3. **Database** might need to act differently on insertion of each entity. For example the process of  updating the `AverageGrade` of a student needs to be **triggered** only when a new grade is inserted.